### ISSUE TITLE
Chunks in storage not being read at start time

### ISSUE THREAD

**Issue by LionelCons** [fluent/fluent-bit#3361]
I have enabled storage and I see some old chunks lying around.

After investigation, here is what seems to happen.

When Fluent Bit is stopped, it flushes chunks in memory to storage, as expected. I see in the logs:

```
2021-04-12T15:19:07.274771+02:00 server fluent-bit: ===== Input =====
2021-04-12T15:19:07.274977+02:00 server fluent-bit: /var/log/audit/audit.log (tail)
2021-04-12T15:19:07.275198+02:00 server fluent-bit: │
2021-04-12T15:19:07.275434+02:00 server fluent-bit: ├─ status
2021-04-12T15:19:07.275660+02:00 server fluent-bit: │  └─ overlimit     : no
2021-04-12T15:19:07.275897+02:00 server fluent-bit: │     ├─ mem size   : 1.2K (1248 bytes)
2021-04-12T15:19:07.276939+02:00 server fluent-bit: │     └─ mem limit  : 0b (0 bytes)
2021-04-12T15:19:07.277207+02:00 server fluent-bit: │
2021-04-12T15:19:07.277659+02:00 server fluent-bit: ├─ tasks
2021-04-12T15:19:07.277897+02:00 server fluent-bit: │  ├─ total tasks   : 0
2021-04-12T15:19:07.278131+02:00 server fluent-bit: │  ├─ new           : 0
2021-04-12T15:19:07.278371+02:00 server fluent-bit: │  ├─ running       : 0
2021-04-12T15:19:07.278618+02:00 server fluent-bit: │  └─ size          : 0b (0 bytes)
2021-04-12T15:19:07.279481+02:00 server fluent-bit: │
2021-04-12T15:19:07.279699+02:00 server fluent-bit: └─ chunks
2021-04-12T15:19:07.279946+02:00 server fluent-bit: └─ total chunks  : 1
2021-04-12T15:19:07.280176+02:00 server fluent-bit: ├─ up chunks  : 1
2021-04-12T15:19:07.280424+02:00 server fluent-bit: ├─ down chunks: 0
2021-04-12T15:19:07.280672+02:00 server fluent-bit: └─ busy chunks: 0
2021-04-12T15:19:07.280934+02:00 server fluent-bit: ├─ size    : 0b (0 bytes)
2021-04-12T15:19:07.281164+02:00 server fluent-bit: └─ size err: 0
```

This matches what can be seen on the filesystem:

```
4 -rw-------   1 root     root         4096 Apr 12 15:19 /var/cache/fluent-bit/var/log/audit/audit.log/60326-1618233546.689110512.flb
```

However, when Fluent Bit starts, the chunk is apparently not read and stays on disk. The logs contain:

```
[2021/04/12 15:19:07] [ info] [engine] started (pid=2486)
[2021/04/12 15:19:07] [ info] [storage] version=1.1.1, initializing...
[2021/04/12 15:19:07] [ info] [storage] root path '/var/cache/fluent-bit'
[2021/04/12 15:19:07] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/04/12 15:19:07] [ info] [storage] backlog input plugin: storage_backlog.12
[2021/04/12 15:19:07] [ info] [input:storage_backlog:storage_backlog.12] queue memory limit: 95.4M
[2021/04/12 15:19:07] [ info] [http_server] listen iface=127.0.0.1 tcp_port=2020
```

Querying the monitoring API, we now see:

```
      "/var/log/audit/audit.log" : {
         "chunks" : {
            "busy" : 0,
            "busy_size" : "0b",
            "down" : 0,
            "total" : 0,
            "up" : 0
         },
         "status" : {
            "mem_limit" : "0b",
            "mem_size" : "0b",
            "overlimit" : false
         }
      },
```

So Fluent Bit does not seem to know about this chunk. Several hours after the restart, the chunk file is still in storage.

FWIW, this is with Fluent Bit 1.7.3 and the configuration includes:

```
[SERVICE]
    Flush            1
    Daemon           Off
    Log_File         /var/log/fluent-bit.log
    Log_Level        info
    Parsers_File     parsers.conf
    Plugins_File     plugins.conf
    storage.path     /var/cache/fluent-bit
    storage.metrics  On
    HTTP_Server      On
    HTTP_Listen      127.0.0.1
    HTTP_Port        2020


[INPUT]
    Name          tail
    Alias         /var/log/audit/audit.log
    Path          /var/log/audit/audit.log
    Path_Key      _path
    Key           raw
    Tag           audit
    DB            /var/db/fluent-bit.db
    storage.type  filesystem
```

**Comment by JeffLuoo** [comment]
The path to the chunk looks weird to me: 
```
/var/cache/fluent-bit/var/log/audit/audit.log/
```
as this should usually be under:
```
/var/cache/fluent-bit/tail.0/
```

What is the environment you are running on?

**Comment by LionelCons** [comment]
If Fluent Bit writes to one location but reads from another (when `Alias` is used) then this smells like a bug...

**Comment by edsiper** [comment]
@LionelCons I tested a solution where we change ```/``` characters by ```.``` as we do for Tag tail expansion. 

wondering if is better to just use internal names of plugins and avoid aliases...

**Comment by LionelCons** [comment]
IMHO, it is safer not to use aliases at all (for storage). This way if aliases are added, removed or changed then Fluent Bit still takes into account the relevant chunks.

**Comment by edsiper** [comment]
got it, fixed in 883507ba
